src/overlays/actors/ovl_En_Ge3/z_en_ge3.c

```diff
@@ -30,37 +57,250 @@ const ActorInit En_Ge3_InitVars = {
+    EnGe3_MoveAndBlink(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/func_80A34B00.s")
+void EnGe3_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnGe3* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/func_80A34B90.s")
+    EnGe3_MaintainCollider(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    EnGe3_MoveAndBlink(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/func_80A34C40.s")
+s32 EnGe3_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
+    EnGe3* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/func_80A34CE4.s")
+    switch (limbIndex) {
+        // Hide swords and veil from object_geldb
+        case GELDB_LIMB_VEIL:
+        case GELDB_LIMB_RIGHT_SWORD:
+        case GELDB_LIMB_LEFT_SWORD:
+            *dList = NULL;
+            return false;
+        // Turn head
+        case GELDB_LIMB_HEAD:
+            rot->x += this->headRot.y;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/func_80A34D68.s")
+        // This is a hack to fix the color-changing clothes this Gerudo has on N64 versions
+        default:
+            OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_ge3.c", 547);
+            switch (limbIndex) {
+                case GELDB_LIMB_NECK:
+                    break;
+                case GELDB_LIMB_HEAD:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 80, 60, 10, 255);
+                    break;
+                case GELDB_LIMB_RIGHT_SWORD:
+                case GELDB_LIMB_LEFT_SWORD:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 140, 170, 230, 255);
+                    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+                    break;
+                default:
+                    gDPPipeSync(POLY_OPA_DISP++);
+                    gDPSetEnvColor(POLY_OPA_DISP++, 140, 0, 0, 255);
+                    break;
+            }
+            CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_ge3.c", 566);
+            break;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/EnGe3_Update.s")
+void EnGe3_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    EnGe3* this = THIS;
+    Vec3f D_80A351C8 = { 600.0f, 700.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/func_80A34E58.s")
+    if (limbIndex == GELDB_LIMB_HEAD) {
+        Matrix_MultVec3f(&D_80A351C8, &this->actor.posRot2.pos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ge3/func_80A35004.s")
+void EnGe3_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    static u64* sEyeTextures[] = {
```

Comment:
I suggest leaving this array right above the function, unless it actually has to be in-function static to match. It looks okay right now, but it could become pretty verbose with the full symbol names. It's also nice to leave it available to other functions, if you wanted to add other draw modes while modding for instance.

In general I recommend to avoid in-function static arrays of functions/pointers for these reasons.